execute.php

```diff
@@ -25,8 +25,20 @@
 require_once('includes/utils.php');
 
 // From where the user *really* comes from.
-if (isset($_SERVER['HTTP_X_FORWARDED_FOR'])) {
-  $requester = $_SERVER['HTTP_X_FORWARDED_FOR'];
+if (
+  $config['misc']['enable_http_x_forwarded_for'] === true
+  &&
+  isset($_SERVER['HTTP_X_FORWARDED_FOR'])
+) {
+  // The user can pass several proxy's, which each one will add its own IP address,
+  //  so we like to take only the first IP address
+  $ips = explode(',', $_SERVER['HTTP_X_FORWARDED_FOR']);
+  $ip = trim($ips[0]);
+  $requester = filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6 | FILTER_FLAG_IPV4);
```

Comment:
There is a `is_valid_ip_address()` function available in `utils` that can be used.